snapcraft.yaml

```diff
@@ -8,7 +8,35 @@ type: base
 build-base: core20
 
 parts:
+  consoleconf-deb:
+    plugin: nil
+    source: https://github.com/CanonicalLtd/subiquity.git
+    source-type: git
+    source-branch: core/focal
+    override-pull: |
+      snapcraftctl pull
+      # install build dependencies
+      export DEBIAN_FRONTEND=noninteractive
+      export DEBCONF_NONINTERACTIVE_SEEN=true
+      sudo -E apt-get build-dep -y ./
+    override-build: |
+      # unset the LD_FLAGS and LD_LIBRARY_PATH vars that snapcraft sets for us
+      # as those will point to the $SNAPCRAFT_STAGE which on re-builds will
+      # contain things like libc and friends that confuse the debian package
+      # build system
+      # TODO: should we unset $PATH to not include $SNAPCRAFT_STAGE too?
+      unset LD_FLAGS
+      unset LD_LIBRARY_PATH
+      # run the real build (but just build the binary package, and don't
+      # bother compressing it too much)
+      dpkg-buildpackage -b -Zgzip -zfast
```

Comment:
I rather assumed if we would be pulling from git that we would be installing the contents directly from git rather than making a package and installing that.  What's the value in the packaging indirection here? /usr/share/doc/$pkg contents?  (/var/lib/dpkg is entirely absent from the snap, so it's not as if there's runtime value in it being added to that database.)  dependency resolution at install time?

Reply:
The value is to ensure that during the switch to this everything is the same. Plus console-conf has a few things packaged using debhelper - i.e. systemd jobs, dist-packages Vs site-packages, etc.

Also note dependencies. When installing these two debs, extra debs are pulled from the archive - probert and urwid.

Later, after this lands we can switch from building deb to just using snapcraft python plugin. And stop having .deb packaging here. But care would need to happen to declare runtime debs, and install systemd units.

Note this is actually similar to snapd snapcraft which also builds deb, and then unpacks it.

Reply:
These are installed whilst the base tarball still has dpkg, later all of that is cleaned out of the chroots prior to packing it as a base.